.github/workflows/turbo-ci.yml

```diff
@@ -9,15 +9,28 @@ on: pull_request
 # pnpm deps are cached by lockfile hash
 
 jobs:
+  wasm-diff:
+    outputs:
+      wasm-diff: ${{ steps.diff.outputs.exit_code }}
+    name: Diff
+    runs-on: ubuntu-latest
+    if: github.event_name == 'pull_request' # only run on PRs
+    steps:
+      - uses: actions/checkout@v4
+      - id: diff # store diff exit code in output
+        run: git diff --exit-code main packages/wasm/crate
+        continue-on-error: true
+
   turbo-compile:
     name: Compile
-    # only run crate compile on pull requests that changed the crate
-    if: ${{ github.event_name == 'pull_request' && contains(github.event.pull_request.changed_files, 'packages/wasm/crate/') }}
+    needs: wasm-diff
+    # only run if wasm crate changed, or dispatch
+    if: github.workflow_dispatch || needs.wasm-diff.outputs.wasm-diff == '1'
```

Comment:
should this be `github.event_name == 'workflow_dispatch'`? see https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions